addons/website/static/src/snippets/s_share/000.js

```diff
@@ -77,7 +76,19 @@ const ShareWidget = publicWidget.Widget.extend({
                 return a + title + c;
             })
             .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
+                // Some pages (/profile/user/ID) don't have an image to share.
+                const ogImageEl = document.querySelector("meta[property='og:image']");
+                if (ogImageEl) {
+                    const media = encodeURIComponent(ogImageEl.content);
+                    return a + media + c;
+                }
+                const params = new URLSearchParams(a);
+                params.delete("media");
+                const paramsString = params.toString();
+                if (paramsString) {
+                    return `?${params.toString()}${c}`;
+                }
+                return "?";
```

Comment:
@Guillaume-gdi Can you test all possible cases of URL and come back to me after that? ^^ This new implementation does not work with `https://www.hello.com/world?media={}#foo`, it probably becomes `https://www.hello.com/world?foo`?

Starting to think a simple replace would be easier as I suggested :) and in one line :wink:  Also here you set `paramsString` to `params.toString()` but still do `params.toString()` afterwards...